SourceCode/src/RayTracer.cpp

```diff
@@ -7,61 +7,49 @@
 #include <optional>
 #include <vector>
 
+#include "tracer/SceneParser.h"
+#include "tracer/Triangle.h"
 #include "tracer/Utils.h"
 #include "tracer/Vector.h"
 
-RayTracer::RayTracer(const Image &image, const Camera &camera, const std::string &pathToScene)
-    : rayUpdateRequired(true), image{image}, camera{camera} {
-  this->updateScene(pathToScene);
+RayTracer::RayTracer(const std::string &pathToScene) : rayUpdateRequired(true) {
+  this->scene = SceneParser::parseScene(pathToScene);
 };
 
-const Image &RayTracer::getImage() const {
-  return this->image;
-}
-
-Image &RayTracer::setImage() {
-  this->rayUpdateRequired = true;
-  return this->image;
-}
-
 const Camera &RayTracer::getCamera() const {
-  return this->camera;
+  return this->scene.getCamera();
 }
 
 Camera &RayTracer::setCamera() {
   this->rayUpdateRequired = true;
-  return this->camera;
-}
-
-void RayTracer::updateScene(const std::string &pathToNewScene) {
-  this->sceneTriangles = this->sceneParser.parseScene(pathToNewScene);
+  return this->scene.setCamera();
 }
 
 void RayTracer::updateRays() {
-  this->pixelRays.resize(this->image.height);
+  this->pixelRays.resize(this->scene.sceneSettings.image.height);
   for (auto &row : this->pixelRays) {
-    row.resize(this->image.width);
+    row.resize(this->scene.sceneSettings.image.width);
   }
-  for (unsigned int pixelRow = 0; pixelRow < this->image.height; pixelRow++) {
-    for (unsigned int pixelCol = 0; pixelCol < this->image.width; pixelCol++) {
+  for (unsigned int pixelRow = 0; pixelRow < this->scene.sceneSettings.image.height; pixelRow++) {
+    for (unsigned int pixelCol = 0; pixelCol < this->scene.sceneSettings.image.width; pixelCol++) {
       float x = static_cast<float>(pixelCol) + 0.5f;
       float y = static_cast<float>(pixelRow) + 0.5f;
 
-      x = x / static_cast<float>(image.width);
-      y = y / static_cast<float>(image.height);
+      x = x / static_cast<float>(this->scene.sceneSettings.image.width);
+      y = y / static_cast<float>(this->scene.sceneSettings.image.height);
 
       x = (2.0f * x) - 1.0f;
       y = 1.0f - (2.0f * y);
 
-      x = x * (static_cast<float>(image.width) / static_cast<float>(image.height));
+      x = x * (static_cast<float>(this->scene.sceneSettings.image.width) /
+               static_cast<float>(this->scene.sceneSettings.image.height));
 
       Vector direction(x, y, -1.0);
-      direction = direction * this->camera.getRotationMatrix();
+      direction = direction * this->scene.getCamera().getRotationMatrix();
       direction.normalize();
-      this->pixelRays[pixelRow][pixelCol] = Ray(this->camera.getPosition(), direction);
+      this->pixelRays[pixelRow][pixelCol] = Ray(this->scene.getCamera().getPosition(), direction);
```

Comment:
There is no need to store the camera rays anymore, you could generate and trace them directly